cpp/csp/adapters/utils/CMakeLists.txt

```
@@ -27,7 +27,13 @@ set_target_properties(csp_adapter_utils PROPERTIES PUBLIC_HEADER "${ADAPTER_UTIL
 find_package(Protobuf REQUIRED)
 target_link_libraries(csp_adapter_utils PRIVATE protobuf::libprotoc protobuf::libprotobuf protobuf::libprotobuf-lite)
 
-install(TARGETS csp_adapter_utils
+add_library(csp_adapter_utils_shared SHARED ${ADAPTER_UTILS_FILES})
+set_target_properties(csp_adapter_utils_shared PROPERTIES PUBLIC_HEADER "${ADAPTER_UTILS_PUBLIC_HEADERS}")
```

this makes an existing situation a bit messier.

static libs are already suffixed by `_static` and prefixed by `lib`. 

```
libbaselibimpl_static.a
libbasketlibimpl_static.a
libcsp_adapter_utils_static.a
libcsp_core_static.a
libcsp_engine_static.a
libcsp_kafka_adapter_static.a
libcsp_parquet_adapter_static.a
libcsp_types_static.a
libcsp_websocket_client_adapter_static.a
libmathimpl_static.a
```

Currently only python-importable shared libraries are exposed, they have prefix `_` and no suffix.

```
_cspbaselibimpl.so
_cspbasketlibimpl.so
_cspimpl.so
_cspmathimpl.so
_cspnpstatsimpl.so
_cspstatsimpl.so
_csptestlibimpl.so
_csptypesimpl.so
_kafkaadapterimpl.so
_parquetadapterimpl.so
_websocketadapterimpl.so
```
So I would probably adopt the `lib` prefix of non-python-importable libraries and maintain the import method suffix

```suggestion
set_target_properties(csp_adapter_utils_shared PROPERTIES PUBLIC_HEADER "${ADAPTER_UTILS_PUBLIC_HEADERS}" PREFIX lib)
```

---

added prefix